pallets/manta-sbt/src/lib.rs

```diff
@@ -98,6 +123,57 @@ const ENCODED_ONE: [u8; 16] = 1u128.to_le_bytes();
 pub type BalanceOf<T> =
     <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 
+/// Eth based address
+type EvmAddress = H160;
+
+/// A signature (a 512-bit value, plus 8 bits for recovery ID).
+pub type Eip712Signature = [u8; 65];
+
+/// Enum with each possible type of allowlisted Eth Address
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum EvmAddressType {
+    /// BAB holder
+    Bab(EvmAddress),
+    /// Galxe holder
+    Galxe(EvmAddress),
+}
+
+/// Different mint types
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum MintType {
+    Bab,
+    Galxe,
+    Manta,
+}
+
+impl From<EvmAddressType> for MintType {
+    fn from(address_type: EvmAddressType) -> Self {
+        match address_type {
+            EvmAddressType::Bab(_) => MintType::Bab,
+            EvmAddressType::Galxe(_) => MintType::Galxe,
+        }
+    }
+}
+
+/// zkSBT mint Status of `EvmAddressType`. This has flag `AlreadyMinted` to put into storage after succesful mint
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum MintStatus {
+    Available(StandardAssetId),
+    AlreadyMinted,
+}
+
+#[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct Metadata<T: Config> {
```

Comment:
maybe this name is kind of general, we already have metadata like asset metadata, runtime metadata.

Reply:
SbtMetadata?